Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/ReflectionAdapter.cpp

```diff
@@ -30,27 +30,128 @@ namespace AZ::DocumentPropertyEditor
 
         struct BoundContainer
         {
-            AZ::SerializeContext::IDataContainer* m_container;
-            void* m_instance;
-            void* m_elementInstance = nullptr;
-            size_t m_elementIndex = 0;
+            // For constructing non-nested containers
+            BoundContainer(AZ::SerializeContext::IDataContainer* container, void* containerInstance)
+                : m_container(container)
+                , m_containerInstance(containerInstance)
+            {
+            }
+
+            // For constructing non-container elements
+            BoundContainer(
+                AZ::SerializeContext::IDataContainer* container,
+                void* containerInstance,
+                void* elementInstance)
+                : m_parentContainer(container)
+                , m_parentContainerInstance(containerInstance)
+                , m_elementInstance(elementInstance)
+            {
+            }
+
+            // For constructing nested containers
```

Comment:
Can you defined the difference between what a nested container is vs a non-nested container vs non-container elements?

Such as 
```c++
//! Nested Container = `AZStd::unordered_map<int, AZStd::vector<int>>`
//! Non-Nested Container = `AZStd::unordered_map<int, int>`
//! Non-Container elemments = `typename AZStd::unordered_map<int, int>::value_type` i.e the type is `AZStd::pair<const int, int>`

Reply:
Will add something like this with the changes we talked about yesterday, even though the cases won't be separated like they are here. Would be good to document the distinction.